code/game/machinery/rechargestation.dm

```diff
@@ -113,11 +112,11 @@
 			R.heal_bodypart_damage(repairs, repairs - 1)
 		if(R.cell)
 			R.cell.charge = min(R.cell.charge + recharge_speed, R.cell.maxcharge)
-	if(isethereal(occupant))
-		var/mob/living/carbon/human/H = occupant
-		var/datum/species/ethereal/E = H.dna?.species
-		if(E)
-			E.adjust_charge(recharge_speed / 70) //Around 3 per process if unupgraded
+	if(iscarbon(occupant) && HAS_TRAIT(occupant, TRAIT_CHARGER))
+		var/mob/living/carbon/C = occupant
+		for(var/V in C.internal_organs)

```

Comment:
don't do this, send a signal that the appropriate organ registers on

Reply:
Give me a minute